src/main/java/org/folio/circulation/resources/RequestByInstanceIdResource.java

```diff
@@ -364,33 +358,63 @@ private void createInstanceLevelRequests(RoutingContext routingContext) {
-    Result<Map<Item, DateTime>> itemDueDateMapResult,
-    Result<Map<Item, Integer>> itemRequestQueueSizeMapResult,
+    Map<Item, DateTime> itemDueDateMap,
     InstanceRequestRelatedRecords records){
 
     return of(() -> {
-      if (itemDueDateMapResult.succeeded() && itemRequestQueueSizeMapResult.succeeded()) {
-        Map<Item, DateTime> itemDueDateMap = itemDueDateMapResult.value();
-        Map<Item, Integer> itemQueueSizeMap = itemRequestQueueSizeMapResult.value();
+        Map<Item, RequestQueue> itemQueueMap = records.getItemRequestQueueMap();
 
-        if (itemDueDateMap == null && itemQueueSizeMap == null) {
+        if (itemDueDateMap == null && itemQueueMap == null) {
           return records;
         }
 
-        //Sort the map
+        //transform itemQueueMap to itemQueueSizeMap
+        Map<Item, Integer> itemQueueSizeMap = itemQueueMap.entrySet()
+          .stream()
+          .collect(Collectors.toMap(
+            Map.Entry::getKey,
+            entry -> entry.getValue().size()
+            ));
+
+        //Sort the itemQueueSize map
         Map<Item, Integer> sortedMap = sortRequestQueues(itemQueueSizeMap, itemDueDateMap,
           records.getInstanceLevelRequest().getPickupServicePointId());
 
         ArrayList<Item> finalOrdedList = new ArrayList<>(sortedMap.keySet());
 
         //put the items that it wasn't able to retrieve loans or RequestQueues for on the bottom of the list
-        finalOrdedList.addAll(records.getItemsWithoutLoans());
-        finalOrdedList.addAll(records.getItemsWithoutRequests());
+        if (records.getItemsWithoutLoans() != null)
+          finalOrdedList.addAll(records.getItemsWithoutLoans());
+        if (records.getItemsWithoutRequests() != null)
+          finalOrdedList.addAll(records.getItemsWithoutRequests());
         records.setSortedUnavailableItems(finalOrdedList);
-      }
       return records;
     });
   }
+
+  private Result<Collection<Item>> validateRequester(Map<Item, RequestQueue> itemRequestQueueMap, UUID requesterId) {
+    if (!itemRequestQueueMap.isEmpty()) {
+      Collection<RequestQueue> requestQueues = itemRequestQueueMap.values();
+      for (RequestQueue queue : requestQueues) {
+        final Optional<Request> matchingRequest = queue.getRequests()
+          .stream()
+          .filter(request -> request.isOpen()
+                           && Objects.equals(request.asJson().getString(REQUESTER_ID_FIELD),
+                                             requesterId.toString())
+          ).findFirst();
+
+        if (matchingRequest.isPresent()) {
+            Map<String, String> parameters = new HashMap<>();
+            parameters.put(REQUESTER_ID_FIELD, requesterId.toString());
+            parameters.put(ITEM_ID_FIELD, matchingRequest.get().asJson().getString(ITEM_ID_FIELD));
+            String message = "This requester already has an open request for a copy of this title";
```

Comment:
This message refers to these as title level requests. In general, are we referring to these as title or instance level requests?

We use instance in the path and  instanceId as an input parameter, I don’t know if we use instance in other messages or in the discovery system presentation (I imagine that is title).

I’m aware there is a tension in the language in this area, and that might not be solvable here, or the PO may have provided this message. Just thought I’d ask.

Reply:
I will double check with the PO on messaging.

Reply:
We refer to those requests as Title Level Requests.  I also think that "title" is more meaningful to patrons than "instance".  We could compromise on: "This requester already has an open request for a copy of this title (instance)"

Reply:
Maybe we can make these FOLIO consistent once this is in place: https://issues.folio.org/browse/UXPROD-266

"copy" vs "item" is another case.

Reply:
Magda, Matt, and I discussed about this at length and Magda decided to go with the FOLIO terminology "instance" and "item" for now, but will create a mapping for messages in EDS in the near future for a more friendly message.